src/CLaSH/Signal/Internal.hs

```diff
@@ -269,7 +269,7 @@ register# _ i s = i :- s
 regEn# :: SClock clk -> a -> Signal' clk Bool -> Signal' clk a -> Signal' clk a
 regEn# _ = go
   where
-    go o (e :- es) as@(~(x :- xs)) =
+    go o ~(e :- es) as@(~(x :- xs)) =
```

Comment:
Can you add the same `@`+`seq` pattern for the enables, otherwise this definition might leak the stream constructors `:-` in the enable:

So:

```haskell
go o en@(e :- es) as@(~(x :- xs)) =
  o `seqX` o :- (en `seq` as `seq` if e then go x es xs else go o es xs)
```